overlord/devicestate/generate_rsa_key.go

```diff
@@ -18,38 +18,45 @@
 
 package devicestate
 
-/*
-#cgo pkg-config: openssl
-#include "rsa_generate_key.h"
-*/
-import "C"
-
 import (
 	"crypto/rsa"
 	"crypto/x509"
 	"encoding/pem"
 	"errors"
-	"unsafe"
+	"io/ioutil"
+	"os"
+	"os/exec"
+	"strconv"
+
+	"github.com/snapcore/snapd/osutil"
+)
+
+var (
+	sshKeyFile       = "snapd.key.tmp"
+	sshPublicKeyFile = sshKeyFile + ".pub"
 )
 
 func generateRSAKey(keyLength int) (*rsa.PrivateKey, error) {
-	var privateKey C.SnapdRSAKeyGenerationBuffer
+	defer func() {
+		os.Remove(sshKeyFile)
+		os.Remove(sshPublicKeyFile)
+	}()
+
+	os.Remove(sshKeyFile)
+	os.Remove(sshPublicKeyFile)
 
-	switch C.snapd_rsa_generate_key(C.uint64_t(keyLength), &privateKey) {
-	case C.SNAPD_RSA_KEY_GENERATION_SEED_FAILURE:
-		return nil, errors.New("cannot generate RSA key: RNG not seeded")
-	case C.SNAPD_RSA_KEY_GENERATION_ALLOCATION_FAILURE:
-		return nil, errors.New("cannot generate RSA key: could not allocate memory")
-	case C.SNAPD_RSA_KEY_GENERATION_KEY_GENERATION_FAILURE:
-		return nil, errors.New("cannot generate RSA key")
-	case C.SNAPD_RSA_KEY_GENERATION_MARSHAL_FAILURE:
-		return nil, errors.New("cannot generate RSA key: could not marshal key")
-	case C.SNAPD_RSA_KEY_GENERATION_SUCCESS:
-		break
+	cmd := exec.Command("ssh-keygen", "-t", "rsa", "-b", strconv.FormatInt(int64(keyLength), 10), "-N", "", "-f", sshKeyFile)
```

Comment:
Both runtime and build time dependency list openssh-client (for Ubuntu at least).